src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(AwaitCompletionProvider), LanguageNames.CSharp)]
+    [ExtensionOrder(After = nameof(KeywordCompletionProvider))]
```

Comment:
Not sure if this is the correct way.

Reply:
As mentioned here https://github.com/dotnet/roslyn/pull/48352#discussion_r597953737, I'd prefer to have a single provider handles it, instead of relies on order and other implementation details.

Reply:
@genlu The way `AbstractKeywordCompletionProvider` is written doesn't allow (or at least not easily) to do so.
So If I'm going to have a single completion provider for that, I'll have to delete the `AwaitKeywordRecommender` and handle everything in the new provider that inherits directly from `LSPCompletionProvider` instead of `AbstractKeywordCompletionProvider`. Also note that it was suggested to have different providers for this in https://github.com/dotnet/roslyn/issues/45368#issuecomment-647939097

Reply:
here's my recommendation:

1. get rid of AwaitKeywordRecommender
2. move it's logic into this completino provider
3. have this completion provider handle not just offering the keyword, but also the new logic to add 'async' as necessary